pkg/util/hostname.go

```diff
@@ -128,6 +142,7 @@ func GetHostname() (string, error) {
 	// FQDN
 	log.Debug("GetHostname trying FQDN/`hostname -f`...")
 	fqdn, err := getSystemFQDN()
+	provider = "os"
```

Comment:
I feel like we should set `fqdn` here to differentiate with `os` line 162.
Also, we should set `provider` to `container` (or something like that) line 154.